packages/discord.js/src/structures/Emoji.js

```diff
@@ -47,13 +47,18 @@ class Emoji extends Base {
     return encodeURIComponent(this.name);
   }
 
-  /**
+/**
    * The URL to the emoji file if it's a custom emoji
    * @type {?string}
-   * @readonly
+   * @param {ImageExtension} extension the extension of emoji
+   * @returns {string}
    */
-  get url() {
-    return this.id && this.client.rest.cdn.emoji(this.id, this.animated ? 'gif' : 'png');
+  url(extension) {
+    let final_result;
+    const extensions = ['webp', 'png', 'jpg', 'jpeg', 'gif'];
+    if (!extensions.includes(extension)) extension = this.animated ? 'gif' : 'png';
+    final_result = this.id && this.client.rest.cdn.emoji(this.id, extension);
+    return final_result;
```

Comment:
```suggestion
    const extensions = ['webp', 'png', 'jpg', 'jpeg', 'gif'];
    if (!extensions.includes(extension)) extension = this.animated ? 'gif' : 'png';
    return this.id && this.client.rest.cdn.emoji(this.id, extension);
```
You're only assigning to `final_result` once.

Reply:
Good job bro